substratools/algo.py

```diff
@@ -182,6 +182,11 @@ def __init__(self, interface, workspace=None, opener_wrapper=None):
             opener.load_from_module(workspace=self._workspace)
         self._interface = interface
 
+    def _assert_output_model_exists(self):
+        path = self._workspace.output_model_path
+        assert not os.path.isdir(path), f'Expected output model file at {path}, found dir'
```

Comment:
Is there a reason for using assert rather than exceptions?

As far as I know, assert should be used to prevent from developper bugs / errors. At the opposite they should not be used to catch user errors.

That's the reason why, I think it would be better here to raise an exception. This could be done by adding a new error in the exceptions module describing this error (`InvalidOutputError`?).

Reply:
I thought of using Exceptions but since they can't be catched and since they are here only for development purposes (to warn the developer that their `save_model` method has an issue) I figured assertion were more to the point.

Reply:
Assert can be deactivated in production (this is done in python by using the -O option when launching a script for instance).

Using Exceptions could also help for the tests: we will be able to check the exception type easily.

So I think as a good practice we should use assertion only to prevent from developper bugs.